src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -111,20 +111,45 @@ public void setup() throws Exception {
 
 	@Test
 	public void testNameFilter() throws Exception {
-		// contract: legacy NameFilter is tested and works
+		// contract: the constructor of the class NameFilter creates and initialises an object.
+
 		CtClass<?> foo = factory.Package().get("spoon.test.filters.testclasses").getType("Foo");
 		assertEquals("Foo", foo.getSimpleName());
 		List<CtNamedElement> elements = foo.getElements(new NameFilter<>("i"));
-		// contract: tests NameFilter constructor for cases other than null
 		assertEquals(1, elements.size());
-		// contract: tests NameFilter constructor for null value case
-		assertThrows(IllegalArgumentException.class, () -> foo.getElements(new NameFilter<>(null)));
-		// contract: tests the getType method of the NameFilter class
-		assertEquals("CtNamedElement",new NameFilter<>("i").getType().getSimpleName());
-		// contract: tests the matches method of the NameFilter class
-		assertTrue(new NameFilter<>("i").matches(elements.get(0)));
 	}
 
+	@Test
+	void testNameFilterMatchesReturnsTrueForMatchingName() {
+		// contract: NameFilter.matches should return true for an element with a matching name
+
+		// arrange
+		String name = "someNiceName";
+		NameFilter<CtNamedElement> nameFilter = new NameFilter<>(name);
+		CtNamedElement elemWithMatchingName = new Launcher().getFactory().createLocalVariable();
+		elemWithMatchingName.setSimpleName(name);
+
+		// act
+		boolean matches = nameFilter.matches(elemWithMatchingName);
+
+		// assert
+		assertTrue(matches);
+	}
```

Comment:
You're free to keep this test if you want, but it doesn't improve line or mutation coverage since the method is exercised in the same way by the original `testNameFilter` test. So, it's a good and isolated test, but it doesn't actually improve any testing metrics.

It's not necessarily a bad thing to have overlapping tests, but for your future PRs you should focus on tests that do improve the metrics, especially the mutation metrics.

Reply:
Let's keep this test as a reference for whoever will be working on FilterTest in future.

Yes, I will be next targeting tests with mutation metrics. 

I also have a doubt regarding mutation coverage, some files have 100% mutation strength but 40-50% mutation coverage, that does that implies?

Reply:
> I also have a doubt regarding mutation coverage, some files have 100% mutation strength but 40-50% mutation coverage, that does that implies?

This doesn't seem to be documented at all, but I'm pretty sure it's like this:

* Mutation coverage = all mutants / killed mutants
* Test strength = mutants in covered code / killed mutants

In other words, test strength shows how good the current tests are at killing mutants, while mutation coverage also includes mutants that exist in code which isn't even covered by the current tests.

Reply:
got it